### PR TITLE
Improve the linux CMake instructions a bit

### LINKED ISSUE #9450: Clarifications for Linux build instructions
### Location

https://github.com/endless-sky/endless-sky/blob/master/docs/readme-cmake.md#linux

### Describe the issue

This issue is regarding compilation on Ubuntu 20.04.06 LTS

1) Ubuntu 20.04.06 LTS is sufficiently out of date, even with updates packages repositories enabled, that configuration with `-DES_USE_SYSTEM_LIBRARIES=OFF` is mandatory.

2) Project configuration fails due to silent dependency of `curl` as part of `vcpkg` configuration step

### Expected content

- [ ] A short note reminding the user to do this should be sufficient. 
- [ ] Add curl to the list of dependencies

### PR DESCRIPTION
**Pull request by quyykk**
**Bugfix:** This PR fixes #9450 

## Fix Details

Improves the CMake docs for Linux a bit:

- Mention the minimum version needed for Debian/Ubuntu
- Add gles preset mention
- Mention `cmake --list-presets`
- Add a short sentence about needing to install additional dependencies when building from source.

### CODE REVIEW

**Comment by shitwolfymakes** [comment]
I'll spin up a VM and reply when I've had time to confirm

**Comment by shitwolfymakes** [comment]
These worked, but it still needs curl added to the dependencies so it doesn't break during configuration

**Comment by shitwolfymakes** [comment]
Perfect, this is ready to be merged

File: docs/readme-cmake.md [hunk 1]
```diff
@@ -49,17 +49,15 @@ If you want to build the libraries from source instead of using Homebrew, you ca
 
 ### Linux
 
-You will need at least CMake 3.21. You can get the latest version from the [offical website](https://cmake.org/download/).
+You can use your favorite package manager to install the needed dependencies. If you're using a slower moving distro like Ubuntu or Debian (or any derivatives thereof), make sure to use at least Ubuntu 22.04 LTS or Debian 12. If you are using an older version of those distributions, see the note below.
```
Comment (warp-core) [inline docs/readme-cmake.md:52]:
```suggestion
You can use your favorite package manager to install the needed dependencies. If you're using a slower moving distro like Ubuntu or Debian (or any derivatives thereof), make sure to use at least Ubuntu 22.04 LTS or Debian 12.
If your distro does not provide up-to-date version of these libraries, you will need to tell CMake to build the libraries from source by passing `-DES_USE_SYSTEM_LIBRARIES=OFF` while configuring. Older versions of Ubuntu and Debian, for example, will need this. Additional dependencies will likely need to be installed to build the libraries from source as well.

In addition to the below dependencies, you will also need CMake 3.21 or newer. You can get the latest version from the [offical website](https://cmake.org/download/).
```

File: docs/readme-cmake.md [hunk 2]
```diff
@@ -49,17 +49,15 @@ If you want to build the libraries from source instead of using Homebrew, you ca
...
 
-**Note**: If your distro does not provide up-to-date version of the needed libraries, you will need to tell CMake to build the libraries from source by passing `-DES_USE_SYSTEM_LIBRARIES=OFF` while configuring.
-
-If you use a reasonably up-to-date distro, then you can use your favorite package manager to install the needed dependencies.
+**Note**: If your distro does not provide up-to-date version of the needed libraries, you will need to tell CMake to build the libraries from source by passing `-DES_USE_SYSTEM_LIBRARIES=OFF` while configuring. You will also need at least CMake 3.21. You can get the latest version from the [offical website](https://cmake.org/download/). Additional dependencies will likely need to be installed to build the libraries from source as well.
```
Comment (warp-core) [inline docs/readme-cmake.md:54]:
```suggestion
```

**Review by warp-core — CHANGES_REQUESTED** [review]
It might also be useful to include some indication of what versions of the dependencies are necessary, or are we saying you have to have the latest version or build from source?

**Review by warp-core — APPROVED** [review]

**Pull request merged**